core/lib/state/src/storage_factory.rs

```diff
@@ -75,6 +79,17 @@ pub enum PgOrRocksdbStorage<'a> {
     Rocksdb(RocksdbStorage),
     /// Implementation over a RocksDB cache instance with in-memory DB diffs.
     RocksdbWithMemory(RocksdbWithMemory),
+    /// In-memory storage snapshot.
+    Snapshot(StorageSnapshot),
+    /// Generic implementation. Should be used for testing purposes only.
+    Boxed(Box<dyn ReadStorage + Send + 'a>),
```

Comment:
Given the amount of variants this enum has at this point, I guess we should consider renaming it from `PgOrRocksdbStorage` to smth else 😅